src/gui/interfaces/iguiapplication.h

```diff
@@ -32,12 +32,17 @@
 
 #include "base/interfaces/iapplication.h"
 
+class DesktopIntegration;
 class MainWindow;
 
 class IGUIApplication : public IApplication
 {
 public:
     virtual ~IGUIApplication() = default;
 
-    virtual MainWindow *mainWindow() = 0;
+    [[nodiscard]] virtual DesktopIntegration *desktopIntegration() = 0;
+    [[nodiscard]] virtual MainWindow *mainWindow() = 0;
```

Comment:
> I think it is only suitable to functions that perform side-effects because only those could be dangerous if coded incorrectly.

IMO, you misinterprete things.
A lot of functions that have a side effect can be used only for this side effect, and the return value can often be neglected, since it provides some additional information. Calling pure getters, on the contrary, makes no sense if you ignore the return value, so a statement like `app()->mainWindow();` is clearly inappropriate in the code.

I agree that this may look redundant if the code changes are reviewed in detail, but it can still be easily overlooked when reviewing large changes. I agree that discarding the getter return value does not do any harm in itself, but it may signal an incorrectly modified code next to it.

Reply:
>IMO, you misinterprete things.

Not at all.

>I think it is only suitable to functions that perform side-effects because only those could be dangerous if coded incorrectly.

I don't think this statement has any problems. I'm not saying "all functions with side-effects should add `nodiscard`" but "I only expect to see `nodiscard` at some function which performs side-effects". You are *extending* it to inappropriate functions where it isn't applicable.

>Calling pure getters, on the contrary, makes no sense if you ignore the return value, so a statement like app()->mainWindow(); is clearly inappropriate in the code.

Inappropriate yet at the same time, no harm.
Here is std library viewpoint on this: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0600r1.pdf
Not that we should follow it, but we should take notes on its rationale.

IMO your example falls into "It should not be added when: not using the return value makes no sense but doesn’t hurt and is usually not an error" part.
Also the rationale for `empty()` and `top()` is very interesting..

>I agree that discarding the getter return value does not do any harm in itself, but it may signal an incorrectly modified code next to it.

How can `nodiscard` help in here? However if you decide it really is useful, then you can keep it.

Reply:
> How can `nodiscard` help in here?

Just like the other compiler hints, isn't that right? It simply won't compile such code, which will draw your attention to it.

Reply:
Anyway removed.

Reply:
>It simply won't compile such code, which will draw your attention to it.

It will still compile successfully if there is no syntax/other errors, compiler is only *encouraged* to emit a warning about it.